src/Executables/FindHorizons/FindHorizons.hpp

```diff
@@ -0,0 +1,325 @@
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
+  static Parallel::iterable_action_return_t apply(
+      db::DataBox<DbTagsList>& box,
+      const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      const Parallel::GlobalCache<Metavariables>& /*cache*/,
+      const ArrayIndex& /*array_index*/, ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) {
+    // For now, set the time to the observation value at which we want to read
+    // data, or to zero if it's determined from the available observations
+    // automatically. The time is currently only used to index interpolations to
+    // the AH finder. We can generalize this to step through the available
+    // observations in the file, handle time-dependent domains, etc. To do that,
+    // the importer will have to send information about the observation value it
+    // read from the file, as well as the available observation IDs/values in
+    // the file.
+    const double observation_time = std::visit(
+        Overloader{
+            [](const double local_obs_value) { return local_obs_value; },
+            [](const importers::ObservationSelector /*local_obs_selector*/) {
+              return 0.;
+            }},
+        db::get<importers::Tags::ObservationValue<OptionTags::VolumeDataGroup>>(
+            box));
+    ::Initialization::mutate_assign<tmpl::list<::Tags::Time>>(
+        make_not_null(&box), observation_time);
+    // Nothing to do to initialize the fields. They will be read from the
+    // volume data file.
+    return {Parallel::AlgorithmExecution::Continue, std::nullopt};
+  }
+};
+
+// Send volume data to the interpolator, which will trigger an apparent horizon
+// find
+template <typename InterpolationTargetTag>
+struct DispatchApparentHorizonFinder {
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            size_t Dim, typename ActionList, typename ParallelComponent>
+  static Parallel::iterable_action_return_t apply(
+      db::DataBox<DbTagsList>& box,
+      const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      Parallel::GlobalCache<Metavariables>& cache,
+      const ElementId<Dim>& element_id, ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) {
+    intrp::interpolate<InterpolationTargetTag>(
+        db::get<::Tags::Time>(box), db::get<domain::Tags::Mesh<Dim>>(box),
+        cache, element_id,
+        db::get<gr::Tags::SpatialMetric<Dim, Frame::Inertial, DataVector>>(box),
+        db::get<gr::Tags::ExtrinsicCurvature<Dim, Frame::Inertial, DataVector>>(
+            box),
+        db::get<domain::Tags::InverseJacobian<Dim, Frame::ElementLogical,
+                                              Frame::Inertial>>(box));
+    return {Parallel::AlgorithmExecution::Continue, std::nullopt};
+  }
+};
+
+}  // namespace Actions
+
+template <size_t Dim>
+struct ComputeHorizonVolumeQuantities
```

Comment:
Not a change request, but perhaps in the future this struct can be moved outside of this executable  (It would be needed if somewhere else somebody wanted to find an AH using spatial metric and extrinsic curvature)